java/dataset/src/test/java/org/apache/arrow/dataset/TestAllTypes.java

```diff
@@ -0,0 +1,262 @@
+        Field.nullablePrimitive("null", ArrowType.Null.INSTANCE),
+        Field.nullablePrimitive("bool", ArrowType.Bool.INSTANCE),
+        Field.nullablePrimitive("int8", new ArrowType.Int(8, true)),
+        Field.nullablePrimitive("int16", new ArrowType.Int(16, true)),
+        Field.nullablePrimitive("int32", new ArrowType.Int(32, true)),
+        Field.nullablePrimitive("int64", new ArrowType.Int(64, true)),
+        Field.nullablePrimitive("uint8", new ArrowType.Int(8, false)),
+        Field.nullablePrimitive("uint16", new ArrowType.Int(16, false)),
+        Field.nullablePrimitive("uint32", new ArrowType.Int(32, false)),
+        Field.nullablePrimitive("uint64", new ArrowType.Int(64, false)),
+        Field.nullablePrimitive("float32", new ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE)),
+        Field.nullablePrimitive("float64", new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)),
+        Field.nullablePrimitive("utf8", ArrowType.Utf8.INSTANCE),
+        Field.nullablePrimitive("binary", ArrowType.Binary.INSTANCE),
+        Field.nullablePrimitive("largeutf8", ArrowType.LargeUtf8.INSTANCE),
+        Field.nullablePrimitive("largebinary", ArrowType.LargeBinary.INSTANCE),
+        Field.nullablePrimitive("fixed_size_binary", new ArrowType.FixedSizeBinary(1)),
+        Field.nullablePrimitive("date_ms", new ArrowType.Date(DateUnit.MILLISECOND)),
+        Field.nullablePrimitive("time_ms", new ArrowType.Time(TimeUnit.MILLISECOND, 32)),
+        Field.nullablePrimitive("timestamp_ms", new ArrowType.Timestamp(TimeUnit.MILLISECOND, null)),
+        Field.nullablePrimitive("timestamptz_ms", new ArrowType.Timestamp(TimeUnit.MILLISECOND, "UTC")),
+        Field.nullablePrimitive("time_ns", new ArrowType.Time(TimeUnit.NANOSECOND, 64)),
+        Field.nullablePrimitive("timestamp_ns", new ArrowType.Timestamp(TimeUnit.NANOSECOND, null)),
+        Field.nullablePrimitive("timestamptz_ns", new ArrowType.Timestamp(TimeUnit.NANOSECOND, "UTC")),
+        Field.nullablePrimitive("duration", new ArrowType.Duration(TimeUnit.MILLISECOND)),
+        Field.nullablePrimitive("decimal128", new ArrowType.Decimal(10, 2, 128)),
+        Field.nullablePrimitive("decimal256", new ArrowType.Decimal(10, 2, 256)),
+        new Field("list", FieldType.nullable(ArrowType.List.INSTANCE),
+            Collections.singletonList(Field.nullable("items", new ArrowType.Int(32, true)))),
+        new Field("largelist", FieldType.nullable(ArrowType.LargeList.INSTANCE),
+            Collections.singletonList(Field.nullable("items", new ArrowType.Int(32, true)))),
+        new Field("fixedsizelist", FieldType.nullable(new ArrowType.FixedSizeList(2)),
+            Collections.singletonList(Field.nullable("items", new ArrowType.Int(32, true)))),
+        structField
+    );
+    VectorSchemaRoot root = VectorSchemaRoot.create(new Schema(fields), allocator);
+    root.allocateNew();
+    root.setRowCount(2);
+
+    root.getVector("null").setNull(0);
```

Comment:
Not for this PR, but is there a way to reduce the boilerplate?
For example in C++ we have `ArrowFromJSON` and `RecordBatchFromJSON` which are quite useful to quickly write example data for tests.

Reply:
I don't think we have this yet, as I've seen other Java PRs create test cases like this. Definitely would be nice to have.

Reply:
I think this would be really helpful. There were some facilities for building a schema from an Avro file so perhaps those could be used (assuming Avro supports all the same data types).